src/main/java/com/vesoft/nebula/jdbc/impl/NebulaDriver.java

```diff
@@ -140,6 +141,50 @@ public Connection connect(String url, Properties connectionConfig) throws SQLExc
         }
     }
 
+    /**
+     * Parse an extended JDBC url supporting the syntax:
+     *
+     *      jdbc:nebula://host:port,host:port,.../graphSpace
+     *
+     * @param url extended url
+     * @return the original url 'jdbc:nebula://graphSpace'
+     * @throws SQLException if the servers are not specified as 'host:port' or 'port' is not an integer
+     */
+    protected String parseAddress(String url) throws SQLException {
+        // jdbc:nebula://ip:port,ip:port,.../graphspace
+
+        int b = url.indexOf("//");
+        int e = url.indexOf("/", b+2);
+        if (e == -1)
+            return url;
+
+        String addresses = url.substring(b+2, e);
+        url = url.substring(0, b+2) + url.substring(e+1);
+
+        String[] addressesList = addresses.split(",");
+        List<HostAddress> customizedAddressList = new ArrayList<>();
+
+        for (String address : addressesList) {
+            String[] addressInfo =  address.split(":");
+            if(addressInfo.length != 2){
+                throw new SQLException(String.format("url [%s] is invalid, please make sure your url match thr format: \"ip:port\".", url));
+            }
+            String ip = addressInfo[0];
+            int port = Integer.parseInt(addressInfo[1]);
+            customizedAddressList.add(new HostAddress(ip, port));
+        }
+
+        // close the current pool that uses '127.0.0.1' (WHY????)
+        this.nebulaPool.close();
```

Comment:
maybe it should be refactored.
If url is the param of `connect`, then when get driver, there's should't init the pool, just init the pool in `connect` function.
Or give the `url` to driver, and init the connection pool in getDriver function.

The default client address `127.0.0.1` is unreasonable.

Reply:
totally agreed, would be better to remove this one.

Reply:
@Nicole00 I'm sorry but i saw line 180 the `customizedAddressList` will replace the poolProperties's `addressList` , so in my opinion the default IP 127.0.0.1 will by replaced by the `customizedAddressList` ?

Reply:
Yeah, it will be replaced by customizedAddressList only  when user call `new NebulaDriver(address)`, but it's not the standard way to use jdbc. Almost usage is `Class.forName()` and it will call the empty construct function, so the address will be 127.0.0.1.

We can remove the customizedAddressList and put the user address in url, which will be more common.